assets/src/apps/delivery/layouts/deck/components/HistoryPanel.tsx

```diff
@@ -0,0 +1,136 @@
+import { selectCurrentActivityId } from '../../../store/features/activities/slice';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { navigateToActivity } from '../../../store/features/groups/actions/deck';
+import { useState, useEffect } from 'react';
+
+interface HistoryEntry {
+  id: string;
+  name: string;
+  timestamp?: any; // not sure if this is needed
+  current?: boolean;
+  selected?: boolean;
+}
+
+interface HistoryPanelProps {
+  items: HistoryEntry[];
+  onMinimize: any; // function?
+  onRestart: any; // function
+}
+
+interface TimeAgoProps {
+  timeStamp: any;
+}
+const TimeAgo: React.FC<TimeAgoProps> = ({ timeStamp }) => {
+  if (!timeStamp) {
+    return <span></span>;
+  }
+
+  const [time, setTime] = useState('');
+  const MillisToDaysHoursMinutesAndSeconds = (millis: any) => {
+    const minutes = Math.floor(millis / 60000);
+    const hours = Math.floor((millis / (1000 * 60 * 60)) % 24);
+    const days = Math.floor(millis / (1000 * 60 * 60 * 24));
+    if (days > 0) {
+      if (days === 1) {
+        return 'a day ago';
+      }
+      return `${days}  days ago`;
+    } else if (hours > 0) {
+      if (hours === 1) {
+        return 'an hour ago';
+      }
+      return `${hours} hours and ${minutes} minutes ago`;
+    } else if (minutes > 0) {
+      if (minutes === 1) {
+        return 'a minute ago';
+      }
+      return `${minutes}  minutes ago`;
+    } else {
+      return 'a few seconds ago';
+    }
+  };
+  const tick = () => {
+    const currentDate = Date.now();
+    const screenVisitedTime = currentDate - timeStamp;
+    const timeTickerText = MillisToDaysHoursMinutesAndSeconds(screenVisitedTime);
+    setTime(timeTickerText);
+  };
+
+  useEffect(() => {
+    setInterval(() => {
```

Comment:
hmm do we really need to live update this? also in the exit method should clearInterval

Reply:
Yeah. I checked the SS and they live update it